transpire/cli/image.py

```diff
@@ -8,9 +11,58 @@ def commands(**kwargs):
 
 
 @commands.command()
-def build(**kwargs):
+@click.option("--push/--no-push", default=False)
+def build(push, **kwargs):
     """build defined images"""
-    raise NotImplementedError("Not yet implemented!")
+    # must be passed kwarg `remote` if --push is set
+    # tags not in .transpire.toml are passed through kwarg `additional_tags`
+
+    # ensure we have buildkit installed
+    if shutil_which("buildctl") is None:
+        raise RuntimeError("You must install Buildkit to use this script.")
+
+    # get remote and additional tags (like one by commit)
+    remote: str = kwargs.get('remote', None)
+    tags: List[str] = kwargs.get('additional_tags', [])
+
+    if (push is None) and (remote is None):
+        raise ValueError(f"Argument remote cannot be None when --push is set")
+
+    # read the dockerfile path, app name, tags, context
+    with open(f".transpire.toml", 'r') as tomlFile:
+        config = parse_toml(tomlFile.read())
+        
+    build_output: List[bytes] = []
+    for build_data in config['build']:
+
+        dockerfile: str = f"./{build_data['dockerfile']}"
+        name: str = build_data['name']
+        tags.extend(build_data['tags'])
+        build_context: str = build_data.get('context', '.')
+    
+        # format the remote location to push to: "{remote}", "{remote}/{name}" or "{name}"
+        repo: str = '/'.join(filter(lambda arg: arg is not None, [remote, name]))
+        if len(repo) < 1:
+            raise Exception(f"Build target has no identifier: \n{build_data}")
```

Comment:
I'm a bit unclear on this. Do we always need a build name? If not, should we require it anyway for consistency? If passed a remote but no name, should we build at all?

Reply:
`name` should be a required field, yes